dbt_sugar/core/connectors/base.py

```diff
@@ -0,0 +1,55 @@
+"""
+Module base connector.
+
+Only use this class implemented by a child connector.
+"""
+import logging
+from typing import Any, List, Optional, Tuple
+
+
+class BaseConnector:
+    """
+    Class base connector.
+
+    Parent class of all the connectors.
+    """
+
+    def generate_connection(self) -> None:
+        """
+        Method that creates the connection.
+
+        :return: the specific connection of each Child.
+        """
```

Comment:
This is a bit nitpicky but I'd like the project to use the google style docstrings.

This is configurable if you use an extension like Python docstrings. 

Alternatively, if you want to write them by hand this is a good reference:
https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html

I will add this guideline also in the `CONTRIBUTING.md` file, sorry that I didn't tell you before.